x/pool-incentives/keeper/grpc_query.go

```diff
@@ -143,54 +143,66 @@ func (q Querier) IncentivizedPools(ctx context.Context, _ *types.QueryIncentiviz
-		// This comes from the superfluid keeper, since superfluid is the only pool incentives connected
-		// module that has access to the gamm modules store.
-		migrationRecords, err := q.gammKeeper.GetAllMigrationInfo(sdkCtx)
+		gauge, err := q.incentivesKeeper.GetGaugeByID(sdkCtx, record.GaugeId)
 		if err != nil {
 			return nil, status.Error(codes.Internal, err.Error())
 		}
 
-		// Iterate over all migration records.
-		for _, record := range migrationRecords.BalancerToConcentratedPoolLinks {
-			// If the cl pool is not in the list of incentivized pools, skip it.
-			lockableDuration, incentivized := incentivizedPoolIDs[record.ClPoolId]
-			if !incentivized {
+		if gauge.DistributeTo.LockQueryType == lockuptypes.ByGroup {
+			group, err := q.Keeper.incentivesKeeper.GetGroupByGaugeID(sdkCtx, record.GaugeId)
+			if err != nil {
+				return nil, status.Error(codes.Internal, err.Error())
+			}
+			groupGauge, err := q.Keeper.incentivesKeeper.GetGaugeByID(sdkCtx, group.GroupGaugeId)
+			if err != nil {
+				return nil, status.Error(codes.Internal, err.Error())
+			}
+			if !groupGauge.IsPerpetual {
+				// if the group is not perpetual, it is an externally incentivized gauge so we skip it
 				continue
 			}
+			poolIds, durations, err := q.Keeper.incentivesKeeper.GetPoolIdsAndDurationsFromGroup(sdkCtx, group)
+			if err != nil {
+				return nil, status.Error(codes.Internal, err.Error())
+			}
+			for i, poolId := range poolIds {
+				incentivizedPool := types.IncentivizedPool{
+					PoolId:           poolId,
+					LockableDuration: durations[i],
+					GaugeId:          record.GaugeId,
+				}
 
-			// Add the indirectly incentivized balancer pools to the list of incentivized pools.
-			gaugeId, err := q.Keeper.GetPoolGaugeId(sdkCtx, record.BalancerPoolId, lockableDuration)
+				incentivizedPools = append(incentivizedPools, incentivizedPool)
+			}
+		} else if gauge.DistributeTo.LockQueryType == lockuptypes.NoLock {
+			poolId, err := q.Keeper.GetPoolIdFromGaugeId(sdkCtx, record.GaugeId, epochDuration)
 			if err == nil {
 				incentivizedPool := types.IncentivizedPool{
-					PoolId:           record.BalancerPoolId,
-					LockableDuration: lockableDuration,
-					GaugeId:          gaugeId,
+					PoolId:           poolId,
+					LockableDuration: epochDuration,
+					GaugeId:          record.GaugeId,
 				}
 
 				incentivizedPools = append(incentivizedPools, incentivizedPool)
 			}
+		} else if gauge.DistributeTo.LockQueryType == lockuptypes.ByDuration {
+			for _, lockableDuration := range lockableDurations {
+				poolId, err := q.Keeper.GetPoolIdFromGaugeId(sdkCtx, record.GaugeId, lockableDuration)
+				if err == nil {
+					incentivizedPool := types.IncentivizedPool{
+						PoolId:           poolId,
+						LockableDuration: lockableDuration,
+						GaugeId:          record.GaugeId,
+					}
+
+					incentivizedPools = append(incentivizedPools, incentivizedPool)
+				}
```

Comment:
This implies that we silently skip the error if it occurs. Why do we do that?

Reply:
This is the same logic that this query had prior to this PR. I think we should not change this here and should instead add an issue to investigate why it was like this prior to this PR. If you agree I can make the issue and link it here.

Reply:
Added https://github.com/osmosis-labs/osmosis/issues/6617